File: Mar-18/17-03-18/python/01-Dedication-level-infinity.py
n = int(input())
ar = {}

for i in range(0,n):
    data = input().split()
    ar[int(data[1])] = data[0]

cnt = 0
for i in reversed(sorted(ar)):
    cnt += 1
    if(cnt<=3):
        print(ar[i])
    else:
        break

File: Mar-18/17-03-18/java/01-Dedication-level-infinity.java
import java.io.*;
import java.util.*;


class TestClass {
    public static void main(String args[] ) throws Exception {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        
        int n = Integer.parseInt(input.readLine());
        int ar[] = new int[n];
        String str[] = new String[1001];
        
        for(int i=0;i<n;i++)
        {
            String data[] = input.readLine().split(" ");
            ar[i] =  Integer.parseInt(data[1]);
            str[ar[i]]=data[0];
        }
        
        Arrays.sort(ar);
        for(int i=n-1;i>n-4;i--)
        {
            System.out.println(str[ar[i]]);
        }
    }
}
